packages/common/src/ui/layout/tables/DataTable.tsx

```diff
@@ -32,10 +33,44 @@ export const DataTableComponent = <T extends RecordWithId>({
   isDisabled = false,
 }: TableProps<T>): JSX.Element => {
   const t = useTranslation('common');
-  const { setActiveRows, setDisabledRows } = useTableStore();
+  const { setRows, setDisabledRows, setFocusUp, setFocusDown } =
+    useTableStore();
+  const [clickFocusedRow, setClickFocusedRow] = useState(false);
```

Comment:
This seems a bit hacky, but needed a way to let each row know when it's been clicked (by keyboard) without registering keyboard shortcut inside every row instance. Better ways to do this?

Reply:
Nice!  I have to admit, I don't fully understand refs, so never quite sure when they're useful. Probably a good time to learn.

For now, I've made a new issue summarising your suggestions in here, will do them together at some point when I have some time to get my head around refs properly :) 

https://github.com/openmsupply/openmsupply-client/issues/1071

Reply:
> I think I like the idea of having a kbar action to focus individual rows (i.e. I could type: `focus 12` and focus row 12. Maybe `click 12` to click row 12. Or maybe one or the other? Not sure if it'd be used by end users, but _I_ think it's wicked 😁

Having dynamic values in a kbar shortcut is *probably* a bit beyond what it can easily do. Not sure how you'd capture key events without telling it in advance what numbers to be watching for. Something to think about...

(Though the rows aren't usually numbered -- so you how are you supposed to know what number to type in without manually counting through them ?? 🤔 )

Reply:
refs are pretty rarely used but i think useful in the rare cases. Essentially a value that is 'state' but not 'react state' (causing re-renders), I guess!